### ISSUE TITLE
ModelMapper JOOQ integration bug

### ISSUE THREAD

**Issue by kubalavl** [modelmapper/modelmapper#118]
Imagine following situation:
- I would like to map JooqRecord (JOOQ generated class from table) to POJO (with getters and setters)
- Everything works fine when first mapped source object is completly filled (every property is not null)
- In case of first mapped source object is not completly filled (some properties contains null values). These (not filled) source properties from first mapping won't be never filled (to appropriate destination properties) for followup mappings with same model mapper instance (even if it should be because of not null source property value)

Sample of such situation (Unexpected assertion fail on row 8):

```
    (1) ModelMapper mapper = new ModelMapper();
    (2) mapper.getConfiguration().addValueReader(new RecordValueReader());

    // create country record with null property name   
    (3) CountryRecord record1 = new CountryRecord(1l, true, null, null, null);
    (4) Country1 c1 = mapper.map(record1, Country1.class);
    (5) Assert.assertEquals(record1.getName(), c1.getName());  // Assertion OK

    // create country record with property name = "XYZ"  
    (6) CountryRecord record2 = new CountryRecord(1l, true, "XYZ", null, null);
    (7) Country1 c2 = mapper.map(record2, Country1.class);
    (8) Assert.assertEquals(record2.getName(), c2.getName());  // Assertion fails "XYZ" != null
```

**Comment by sandypad** [comment]
I am facing a similar issue and I was wondering if this bug is fixed yet? Appreciate your help.
Thanks.

**Comment by Roger16** [comment]
I am facing a similar issue and I was wondering if this bug is fixed? I use the general Record object. Appreciate your help.

**Comment by pspi** [comment]
I'm facing this as well. Any updates on how to fix or work around this?

**Comment by jhalterman** [comment]
Related to #168.

**Comment by chhsiao90** [comment]
Fixed by #210
